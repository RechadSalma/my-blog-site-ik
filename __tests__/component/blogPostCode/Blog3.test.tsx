import { render } from "@testing-library/react";
import Blog3Codes from "../../../components/blogPostCodes/Blog3";

describe("<Blog3/> code for blog3 component", () => {
  const blogThreeComponent = () => render(<Blog3Codes />);

  test("renders the <code> tag", () => {
    const { getByRole } = blogThreeComponent();

    const codeTag = getByRole("code");

    expect(codeTag).toBeInTheDocument();
  });

  // snapshot component
  test("snapshot Blog3 code tag", () => {
    const { container } = blogThreeComponent();

    expect(container).toMatchSnapshot();
  });
});
