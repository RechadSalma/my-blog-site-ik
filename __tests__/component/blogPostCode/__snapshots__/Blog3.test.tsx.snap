// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Blog3/> code for blog3 component snapshot Blog3 code tag 1`] = `
<div>
  <pre
    class="preCode mx-auto"
  >
    <code
      class="codeTag mx-auto"
      role="code"
    >
      <div>
        <span
          class="keyword"
        >
          import
        </span>
         React, { ChangeEvent, FormEvent, useState } 
        <span
          class="keyword"
        >
          from
        </span>
         "react";
      </div>
      <br />
      <div>
        <span
          class="keyword"
        >
          export default function
        </span>
         FormControl() {
      </div>
      <div
        class="tab1 comment"
      >
        // For the change event
      </div>
      <div
        class="tab1"
      >
        <span
          class="keyword"
        >
          const
        </span>
         [changeValue, setChangeValue] = 
        <span
          class="method"
        >
          useState
        </span>
        &lt;string&gt;(
        <span
          class="value"
        >
          "initial value"
        </span>
        );
      </div>
      <br />
      <div
        class="tab1 comment"
      >
        // For the submit event
      </div>
      <div
        class="tab1"
      >
        <span
          class="keyword"
        >
          const
        </span>
         [submitValue, setSubmitValue] = 
        <span
          class="method"
        >
          useState
        </span>
        &lt;string&gt;(
        <span
          class="value"
        >
          ""
        </span>
        );
      </div>
      <br />
      <div
        class="tab1 comment"
      >
        // When user submit form
      </div>
      <div
        class="tab1"
      >
        <span
          class="keyword"
        >
          const
        </span>
         submittedForm = (event: 
        <span
          class="keyword"
        >
          FormEvent
        </span>
        &lt;HTMLFormElement&gt;) =&gt; {
      </div>
      <div
        class="tab2"
      >
        <span
          class="comment"
        >
          // Prevent page reloading
        </span>
        <div>
          event
          <span
            class="method"
          >
            .preventDefault
          </span>
          ()
        </div>
        <div>
          <span
            class="comment"
          >
            // Collects the textarea element
          </span>
        </div>
        <div>
          <span
            class="keyword"
          >
            const
          </span>
           textarea = event
          <span
            class="method"
          >
            .currentTarget.children
          </span>
          [0] 
          <span
            class="keyword"
          >
            as
          </span>
           HTMLTextAreaElement; 
        </div>
        <span
          class="comment"
        >
          // Set the final value
        </span>
        <div>
          setSubmitValue(textarea
          <span
            class="method"
          >
            .value
          </span>
          );
        </div>
      </div>
      <div
        class="tab1"
      >
        };
      </div>
      <br />
      <div
        class="tab1"
      >
        <div>
          <span
            class="comment"
          >
            //When user types within the textarea
          </span>
        </div>
        <div>
          <span
            class="keyword"
          >
            const
          </span>
           changeTextarea = (event: 
          <span
            class="keyword"
          >
            ChangeEvent
          </span>
          ) =&gt; {
        </div>
      </div>
      <div
        class="tab2"
      >
        <div
          class="comment"
        >
          // Collects the textarea element
        </div>
        <div>
          <span
            class="keyword"
          >
            const
          </span>
           textarea = event
          <span
            class="method"
          >
            .currentTarget
          </span>
           
          <span
            class="keyword"
          >
            as
          </span>
           HTMLTextAreaElement;
        </div>
        <span
          class="comment"
        >
          // Update the textarea element value with the user inputs
        </span>
        <div>
          setChangeValue(textarea
          <span
            class="method"
          >
            .value
          </span>
          );
        </div>
      </div>
      <div
        class="tab1"
      >
        };
      </div>
      <br />
      <div
        class="tab1"
      >
        <div>
          <span
            class="keyword"
          >
            return
          </span>
           (
        </div>
      </div>
      <div
        class="tab2"
      >
        <div>
          &lt;div&gt;
        </div>
      </div>
      <div
        class="tab3"
      >
        <div>
          &lt;form 
          <span
            class="keyword"
          >
            onSubmit
          </span>
          ={submittedForm}&gt;
        </div>
      </div>
      <div
        class="tab4"
      >
        <div>
          &lt;textarea
        </div>
      </div>
      <div
        class="tab5"
      >
        <div>
          <span
            class="keyword"
          >
            name
          </span>
          =
          <span
            class="value"
          >
            "textareaName"
          </span>
        </div>
        <div>
          <span
            class="keyword"
          >
            value
          </span>
          ={changeValue}
        </div>
        <div>
          <span
            class="keyword"
          >
            onChange
          </span>
          ={changeTextarea}
        </div>
      </div>
      <div
        class="tab4"
      >
        /&gt;
      </div>
      <div
        class="tab4"
      >
        &lt;button 
        <span
          class="keyword"
        >
          type
        </span>
        =
        <span
          class="value"
        >
          "submit"
        </span>
        &gt;Submit&lt;/button&gt;
      </div>
      <div
        class="tab3"
      >
        <div>
          &lt;/form&gt;
        </div>
        <br />
        <div>
          &lt;div&gt;{submitValue}&lt;/div&gt;
        </div>
      </div>
      <div
        class="tab2"
      >
        &lt;/div&gt;
      </div>
      <div
        class="tab1"
      >
        );
      </div>
      <div>
        };
      </div>
      <br />
    </code>
  </pre>
</div>
`;
